rollout/pause.go

```diff
@@ -185,7 +185,10 @@ func (pCtx *pauseContext) CompletedCanaryPauseStep(pause v1alpha1.RolloutPause)
 	rollout := pCtx.rollout
 	pauseCondition := getPauseCondition(rollout, v1alpha1.PauseReasonCanaryPauseStep)
 
-	if pause.Duration != nil {
+	if rollout.Status.ControllerPause && pauseCondition == nil {
+		pCtx.log.Info("Rollout has been unpaused")
+		return true
+	} else if pause.Duration != nil {
```

Comment:
Can you describe the bug which made this fix necessary? If it's reproducible can we modify an e2e test to catch the bug?

Reply:
if the step is pause with duration, then if we do promote, it wont get promoted as expected as the first condition is checking for duration and if not duration met, it just returns without checking for pause condition.
checking for pause condition first will fix the issue.. i have a test that validated the use case.

and in promote command, if we do step increment always for canary, then this code is redundant..